Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/validation/validation.go

```diff
@@ -2060,7 +2060,44 @@ func ValidateSecurityContextConstraints(scc *api.SecurityContextConstraints) err
 		//good types
 	default:
 		msg := fmt.Sprintf("invalid strategy type.  Valid values are %s, %s", api.SELinuxStrategyMustRunAs, api.SELinuxStrategyRunAsAny)
-		allErrs = append(allErrs, errs.NewFieldInvalid("seLinuxContext.type", scc.RunAsUser.Type, msg))
+		allErrs = append(allErrs, errs.NewFieldInvalid("seLinuxContext.type", scc.SELinuxContext.Type, msg))
+	}
+
+	// ensure the fsgroup strat has a valid type
+	switch scc.FSGroup.Type {
+	case api.FSGroupStrategyMustRunAs, api.FSGroupStrategyRunAsAny:
+		//good types
+	default:
+		msg := fmt.Sprintf("invalid strategy type.  Valid values are %s, %s", api.FSGroupStrategyMustRunAs, api.FSGroupStrategyRunAsAny)
+		allErrs = append(allErrs, errs.NewFieldInvalid("fsGroup.type", scc.FSGroup.Type, msg))
+	}
+	allErrs = append(allErrs, validateIDRanges(scc.FSGroup.Ranges).Prefix("fsGroup.ranges")...)
+
+	switch scc.SupplementalGroups.Type {
+	case api.SupplementalGroupsStrategyMustRunAs, api.SupplementalGroupsStrategyRunAsAny:
+	//good types
+	default:
+		msg := fmt.Sprintf("invalid strategy type.  Valid values are %s, %s", api.SupplementalGroupsStrategyMustRunAs, api.SupplementalGroupsStrategyRunAsAny)
+		allErrs = append(allErrs, errs.NewFieldInvalid("supplementalGroups.type", scc.SupplementalGroups.Type, msg))
+	}
+	allErrs = append(allErrs, validateIDRanges(scc.SupplementalGroups.Ranges).Prefix("supplementalGroups")...)
+
+	return allErrs
+}
+
+// validateIDRanges ensures the range is valid.
+func validateIDRanges(rng []api.IDRange) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+
+	for i, r := range rng {
+		if r.Min > r.Max {
+			field := fmt.Sprintf("ranges[%d]", i)
```

Comment:
Nevermind, I get it after we chatted about it, I think a comment would help here.